tests/requirements.txt

```diff
@@ -1,8 +1,152 @@
-jinja2
-pyyaml
-resolvelib < 1.1.0
-sphinx == 5.3.0
-sphinx-notfound-page
-sphinx-ansible-theme
-rstcheck < 6  # rstcheck 6.x has problem with rstcheck.core triggered by include files w/ sphinx directives https://github.com/rstcheck/rstcheck-core/issues/3
-antsibull-docs == 2.3.0  # currently approved version
+#
+# This file is autogenerated by pip-compile with Python 3.11
```

Comment:
@felixfontein What are your thoughts on this one?

Looking at the [support matrix](https://docs.ansible.com/ansible-core/devel/reference_appendices/release_and_maintenance.html#ansible-core-support-matrix), 3.10 appears to be a good option that spans all the stable versions and will also be around for core 2.16.

I'd like to enable Dependabot and it would be straightforward enough to get a dependency tree on Ubuntu. But could we run into any issues with the likes of antsibull docs? Could there be a better option? Thanks.

Reply:
@webknjaz Cheers for the FYI on different OS combos. I think we should stick with Python 3.11 because all the build jobs for docs.ansible.com use that version and it works fine. I also generated the dependency tree on an Ubuntu VM and got the same result as Fedora.

Reply:
Maybe we should have a script for updating this file in a reproducable way, like by using a Docker/Podman container with the `python:3.11` image?

Reply:
> Maybe we should have a script for updating this file in a reproducable way, like by using a Docker/Podman container with the `python:3.11` image?

I think, this should be in a follow-up and also integrated in tox/nox. Additionally, it would be really cool to have a matrix of constraints and have the command wrappers pick up the right thing for the current system but that's a non-trivial thing to achieve. In my above Cheroot experiment, I've gone for several components: a pip wrapper that is integrated into tox and auto-injects constraints for the correct env if they exist; and a GHA job generating those pins in separate envs.

Reply:
I don't think we need all of this complication. Having a single constraints file generated with python3.10 (i.e. the lowest version supported on ansible-core's `devel` branch) on Linux should accomplish what we need.

Reply:
@gotmax23 Let's go with the python3.10 constraints. I should've stuck with that from my previous comment.

For the rest I'll create a separate issue to capture some of the different strands in this PR. I really like the suggestion from @felixfontein to have a script or something and getting this complexity out of the way of contributors. And while I highly value all the insights from @webknjaz (integration with tox/nox is spot on), having a matrix for diff build environments seems like too much cowbell for a docs build. Although, yeah, it'd be cool to have.

Reply:
Agreed. I wasn't saying that it's a “must have”. I was just describing what's possible in extreme cases. I also think that for docs, a small amount of constraints is preferable and the contributor guidelines should stress what runtimes are expected. Plus, things like tox/nox could error out or print a warning if the runtime isn't matching the supported list.
Still, I foresee two main categories of contributors using macOS and Linux. So having decided on one CPython runtime (3.10) is helpful, but probably warrants two separate constraints files to account for the conditional transitive deps. Having one just with Linux in mind is a good start and the other one could be added once somebody hits a problem it would address.